test/battle-test.mocha.js

```diff
@@ -0,0 +1,98 @@
+const fs = require('fs');
+const chai = require('chai');
+
+window = global;
+document = window.document = {};
+
+// Without making these modules, the best we can do is directly include them into this workspace.
+eval('' + fs.readFileSync(`${__dirname}/../js/battle-scene-stub.js`));
+eval('' + fs.readFileSync(`${__dirname}/../js/battle-dex.js`));
+eval('' + fs.readFileSync(`${__dirname}/../js/battle-dex-data.js`));
+eval('' + fs.readFileSync(`${__dirname}/../js/battle.js`));
+
+describe('Battle', function () {
+
+	it('should instantiate without issue', function () {
+		var battle = new Battle();
+	});
+
+	it('should process a bunch of messages properly', function () {
+		var battle = new Battle();
+		battle.debug = true;
+
+		var run = function (msg) {
+			battle.add(msg);
+			battle.fastForwardTo(-1);
+		};
```

Comment:
Is there any reason you're adding these one by one instead of all at once?

Reply:
There's no reason... fixed.